sound/soc/intel/boards/sof_es8336.c

```diff
@@ -158,10 +135,10 @@ static int sof_es8316_speaker_power_event(struct snd_soc_dapm_widget *w,
 	struct snd_soc_card *card = w->dapm->card;
 	struct sof_es8336_private *priv = snd_soc_card_get_drvdata(card);
 
-	if (priv->speaker_en == !SND_SOC_DAPM_EVENT_ON(event))
+	if (priv->speaker_en == SND_SOC_DAPM_EVENT_ON(event))
 		return 0;
```

Comment:
Remove the check here and you can solve the 'first time start-up' issue

Reply:
Makes sense. Added a patch at the end of the series dropping the above code.